### PR TITLE
Clean up the markdown in considerations

### PR DESCRIPTION
**Pull request by sudo-bmitch**
Fix spelling, markdown spacing, consistent bullets, code block language hints, and move links to the end.

### CODE REVIEW

File: considerations.md [hunk 1]
```diff
@@ -134,9 +131,15 @@ The production "component" is one or more lowercase letters.
 A "path" is then at least one component, possibly followed by zero or more slash-component pairs.
 The above can be converted into the following regular expression:
 
-```
+```text
 [a-z]+(?:/[a-z]+)*
 ```
```
Comment (tianon) [inline considerations.md:136]: Ooh, I love explicit types on these -- should this one be `regex`/`regexp`? :eyes: (it's not much, but it's something)

```regex
[a-z]+(?:/[a-z]+)*
```

````suggestion
```regex
[a-z]+(?:/[a-z]+)*
```
````

**Review by tianon — APPROVED** [review]

File: considerations.md (hunk 1 above)
Comment (sudo-bmitch) [reply considerations.md:136]: I was looking for that in the github syntax highlighting and couldn't find it. Apparently it uses https://highlightjs.org/static/demo/. Is regex/regexp supported in other markdown renderers?

Comment (tianon) [reply considerations.md:136]: :shrug: worst case they render it as plain text, right?  I don't think I've seen one that renders something unknown with an error or in some unexpected way :sweat_smile:

(I think that https://github.com/github-linguist/linguist/blob/master/lib/linguist/languages.yml is the GitHub set of supported languages, which is where I found https://github.com/github-linguist/linguist/blob/e855ef2b6f90c34074061a2e17acbe853e61b483/lib/linguist/languages.yml#L5859-L5870)

Comment (sudo-bmitch) [reply considerations.md]: Looks good. Fixed!

**Review by tianon — APPROVED** [review]

**Review by vbatts — APPROVED** [review]

**Pull request merged**